build/Extensions/DepsJsonExtensions.cs

```diff
@@ -68,6 +72,99 @@ public static void RemoveOpenTelemetryLibraries(this JsonObject depsJson)
         }
     }
 
+    public static void RemoveOpenTelemetryLibraries(this JsonObject depsJson)
+    {
+        RemoveLibrary(depsJson, lib => lib.StartsWith("OpenTelemetry"));
+    }
+
+    public static async Task CleanDuplicatesFromDepsJsonAsync(this JsonObject depsJson)
+    {
+        var map = DependencyAnalyzer.Analyze(depsJson.GetDependencies());
+
+        var result = AnalyzeAdapterDependencies(
+            // TODO: Scan these packages from OpenTelemetry.AutoInstrumentation.AdditionalDeps
+            await NugetPackageHelper.GetPackageDependenciesAsync("MongoDB.Driver.Core.Extensions.DiagnosticSources", "1.3.0"),
+
+            // TODO: Extract metadata from MongoClientIntegration.cs and source-link it to nuke.build?
+            await NugetPackageHelper.GetPackageDependenciesAsync("MongoDB.Driver", "2.13.3", "2.65535.65535")
+        );
+
+        foreach (var framework in result)
+        {
+            var latestVersion = framework.Value
+                .OrderByDescending(x => x.Key.Version)
+                .FirstOrDefault();
+
+            if (latestVersion.Value != null)
+            {
+                foreach (var dependency in latestVersion.Value)
+                {
+                    // All package versions contain the same optimisable dependency
+                    var canBeOptimized = framework.Value
```

Comment:
Correct, currently it is testing just one removal. I got here confused, how to do the removal, since the framework values here are .net standard 2.0 and .net standard 2.1 but our values are .net 5.0 and .net 6.0.

Reply:
Update: this framework matching seems still incorrect. Installing `MongoDB.Driver.Core.Extensions.DiagnosticSources` .NetStandard 2.0 doesn't mean that `MongoDB.Driver` will be installed .Netstandard 2.0, in reality it is actually .NetStandard 2.1. 

So in reality, first we need to determine what will be really installed for specific .net version. Then find the common parts.